nncf/common/insertion_point_graph.py

```diff
@@ -314,3 +405,39 @@ def get_pre_hook_node_key(node_key: str, input_port_id: int = 0) -> str:
     @staticmethod
     def get_post_hook_node_key(node_key: str) -> str:
         return InsertionPointGraph.POST_HOOK_ID_PREFIX + node_key
+
+
+class ConstantNodesFilter:
+    @staticmethod
+    def filter(ip_graph: InsertionPointGraph, quantizable_layer_node_keys: List[str]) -> InsertionPointGraph:
+        """
+        Removes all Constant nodes from InsertionPointGraph, making it inference graph.
+
+        :param ip_graph: The original InsertionPointGraph.
+        :return: InsertionPointGraph without Constant nodes.
+        """
+
+        def traverse_function(curr_node, output, visited_nodes):
+            if curr_node in visited_nodes:
+                return True, output
+            output.append(curr_node)
+            visited_nodes.append(curr_node)
+            return False, output
+
+        input_nodes = ip_graph.get_input_nodes()
+        if not input_nodes:
+            # Skip for tests where there is no input node.
```

Comment:
Why we should add the functionality in the common part of the framework that uses only in tests?

Reply:
Remove the comment, add `nncf_logger.debug()` message about it. Tests or not, we will have to handle such a case anyway unless we guarantee that the input graph object always has input nodes. Right now, there is no such guarantee.

Reply:
Guys, I can remove the comment and add debug message. 
But what should I do with the models without NNCF_INPUT_NODE? 
I start graph traversing from them and mark all visited node if any node wasn't marked I remove it from IP graph.
How can I adjust this logic for case when there is no NNCF_INPUT_NODE?

Reply:
Keep the logic, test for it.

Reply:
Done